graphql/src/store/resolver.rs

```diff
@@ -356,144 +217,24 @@ where
 
     fn resolve_objects(
         &self,
-        parent: &Option<q::Value>,
+        objects_value: Option<q::Value>,
         field: &q::Field,
-        field_definition: &s::Field,
+        _field_definition: &s::Field,
         object_type: ObjectOrInterface<'_>,
-        arguments: &HashMap<&q::Name, q::Value>,
-        types_for_interface: &BTreeMap<Name, Vec<ObjectType>>,
-        max_first: u32,
+        _arguments: &HashMap<&q::Name, q::Value>,
     ) -> Result<q::Value, QueryExecutionError> {
-        if Self::was_prefetched(parent) {
-            return self.resolve_objects_prefetch(parent, field, object_type);
-        }
-
-        let object_type = object_type.into();
-        let mut query = build_query(
-            object_type,
-            self.block,
-            arguments,
-            types_for_interface,
-            max_first,
-        )?;
-
-        // Add matching filter for derived fields
-        let derived_from_field = sast::get_derived_from_field(object_type, field_definition);
-        let is_derived = derived_from_field.is_some();
-        if let Some(derived_from_field) = derived_from_field {
-            Self::add_filter_for_derived_field(&mut query, parent, derived_from_field);
-        }
-
-        // Return an empty list if we're dealing with a non-derived field that
-        // holds an empty list of references; there's no point in querying the store
-        // if the result will be empty anyway
-        if !is_derived
-            && parent.is_some()
-            && Self::references_field_is_empty(parent, &field_definition.name)
-        {
-            return Ok(q::Value::List(vec![]));
-        }
-
-        // Add matching filter for reference fields
-        if !is_derived {
-            Self::add_filter_for_reference_field(&mut query, parent, field_definition, object_type);
-        }
-
-        let mut entity_values = Vec::new();
-        for entity in self.store.find(query)? {
-            entity_values.push(entity.into())
-        }
-        Ok(q::Value::List(entity_values))
+        self.resolve_objects_prefetch(objects_value, field, object_type)
```

Comment:
You could inline this here -  I only made that a separate method to help readability.